pkg/ostreeuploader/push.go

```diff
@@ -391,12 +392,14 @@ func waitForCheck(status *CheckStatus) *CheckReport {
 	}
 }
 
-func checkRepo(objs map[string]uint32, url *url.URL, token string) map[string]uint32 {
+func checkRepo(objs map[string]uint32, url *url.URL, token string, corId string) map[string]uint32 {
 	jsonObjects, _ := json.Marshal(objs)
 	req, err := http.NewRequest("GET", url.String(), bytes.NewBuffer(jsonObjects))
 	if err != nil {
 		log.Fatalf("Failed to create a request to check objects presence: %s\n", err.Error())
 	}
+	req.Header.Set("X-Correlation-ID", corId)
+	req.Header.Set("X-Request-ID", uuid.New().String())
```

Comment:
couldn't we drop the x-request-id and just automatically generate those on the server side? that seems more fool-proof than trusting the client side, right?

Reply:
Technically we can set it on a server-side, although usually it's used in the way that a client sets the header value, afaik.
In our case, there is nginx between the client and the oshub, so this ID might be useful in terms of troubleshooting/reading nginx logs if a request gets stuck at nginx level.